File: 1-two-sum/two-sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index1 = 0
        while True:
            for index2 in range(index1+1,len(nums)):
                if nums[index1] == target - nums[index2]:
                    return [index1,index2]       
            index1+=1
            

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer, Integer> hashmap = new HashMap<>();
        for(int i=0;i<n;i++){
            int complement = target - nums[i];
            if(hashmap.containsKey(complement)){
                return new int[] {hashmap.get(complement),i};
            }
            hashmap.put(nums[i],i);
        }
        return new int[]{};
    }
}